andela/ah-codeofduty
# #161255362 Users should be able to like and or dislike articles

#### What does this PR do?
* Users should be able to like and or dislike articles

#### Description of tasks completed
* Tests for the like and dislike articles.
* Added the like_articles and dislike_articles fields in the models.
* Added the like and dislike count in the ArticleSerializer.

#### Working endpoints 
* POST `api/articles/<slug>/like/` - adding a like/dislike
* DELETE `api/articles/<slug>/like/ ` - removing a like/dislike

#### How can this PR be manually tested?
* clone the branch.
* To like an article: `POST api/articles/<slug>/like/` with body
`{
"likes":true
}`
* To dislike an article: `POST api/articles/<slug>/like/` with body
`{
"likes":false
}`
* To delete a like/dislike to an article: `DELETE api/articles/<slug>/like/`

* run tests using python manage.py test. All tests should pass.

#### What are the relevant pivotal tracker stories?
* #161255362 - [Users should be able to like and or dislike articles](https://www.pivotaltracker.com/story/show/161255362)

#### Screenshots
###### 1. like article
<img width="1362" alt="screenshot 2018-11-15 at 12 11 27" src="https://user-images.githubusercontent.com/39617195/48542202-9d50e980-e8cf-11e8-9bd4-e87794c97df6.png">

###### 3. dislike article
<img width="1360" alt="screenshot 2018-11-15 at 12 12 30" src="https://user-images.githubusercontent.com/39617195/48542260-c2455c80-e8cf-11e8-92cd-931e94c2a295.png">

###### 4. delete reactions(like/dislike) to article
<img width="1358" alt="screenshot 2018-11-15 at 12 14 15" src="https://user-images.githubusercontent.com/39617195/48542358-03d60780-e8d0-11e8-9a26-9426ade5ccae.png">
